### PR TITLE
New API preview version for ACS Email Inline Attachment feature

### PR DESCRIPTION
**Pull request by natekimball-msft**
# Data Plane API Specification Update Pull Request

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## API Info: The Basics

Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [X] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document: [ACS Email - CID Embedded Inline Images](https://microsoft.sharepoint-df.com/:w:/t/IC3SDK-EmailasaService/EdfX30MljCBOui5fLRx3So8BKPSReTrww3iPw-pUxqOeyA?e=cPoBwu)
* Previous API Spec Doc: N/A ([Original ACS Email REST API Design Doc](https://microsoft.sharepoint-df.com/:w:/r/teams/IC3SDK-EmailasaService/_layouts/15/Doc.aspx?sourcedoc=%7B80A5A4F8-5DD1-436F-99A8-E1C2794FF0A1%7D&file=ACSEmailRestAPISpec.docx&action=default&mobileredirect=true&wdsle=0&share=IQH4pKWA0V1vQ5mo4cJ5T_ChAf3pyhoXqJugFZs9qZWKdnA))
* Updated paths:  `/emails:send`, specifically, the addition of `contentId` to the `EmailAttachment` definition, and updated description for `attachments` in `EmailMessage` definition. The `SendEmail.json` example has also been updated with sample usage of an inline image.

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Schedule a data plane REST API spec review](https://aka.ms/azsdk/onboarding/restapischedule)

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories") 
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by mikeharder** [comment]
I believe the failure in check `Breaking Change(Cross-Version)` is caused by your examples setting header `retry-after` to an `int` when the schema expects a `string`.  Since the bug is already in your examples in main, you will need to open another PR to fix your examples in main, and then the check will run successfully in this PR.

File: specification/communication/data-plane/Email/preview/2024-07-01-preview/examples/GetOperationStatusReturnsNonTerminalStatus.json
Comment (mikeharder) [inline specification/communication/data-plane/Email/preview/2024-07-01-preview/examples/GetOperationStatusReturnsNonTerminalStatus.json:11]:
```suggestion
        "retry-after": "100"
```

File: specification/communication/data-plane/Email/preview/2024-07-01-preview/examples/SendEmail.json
Comment (mikeharder) [inline specification/communication/data-plane/Email/preview/2024-07-01-preview/examples/SendEmail.json:82]:
```suggestion
        "retry-after": "20"
```

File: specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json
Comment (mikeharder) [inline specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: Are you sure this should be `integer` instead of `string`?  The HTTP spec seems to require `string` to support date-based values.

https://www.rfc-editor.org/rfc/rfc9110#field.retry-after

Comment (natekimball-msft) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: I can check the historical reasoning of this choice with the team and get back to you. That being said, I have a couple of follow up questions here: 

If it's the spec that is incorrect, I'm curious as to why these errors are only manifesting in our builds now, despite using the same version of autorest? It appears that it wasn't caught in our team's previous PRs. Examples: [Add base64 formatting to contentInBase64 property by kagbakpem · Pull Request #24617 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/24617), [Azure Communication Services Email API for GA by apattath · Pull Request #22979 · Azure/azure-rest-api-specs (github.com)](https://github.com/Azure/azure-rest-api-specs/pull/22979).
 
If I make changes to the older specs retroactively, are there any considerations here with introducing breaking changes? It seems like in our most recent PR, a tag was added to approve a breaking change (unsure if it was for the same reason). Is that appropriate here?

Comment (mikekistler) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: For breaking changes, we regard the service behavior as the "source of truth". If the API definition changes to more accurately describe the service behavior, we allow this and actually encourage it.

Comment (mikeharder) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: The check also failed in PR 22979.  For PR 24617, the check was bypassed using label `Approved-BreakingChange`, and we no longer have the build log, but I suspect it also failed with the same error.

I think the first question here, is how should `retry-after` be expressed in services, specs, and examples?  Always a `string`, or can each service decide between `string` and `integer`?

For instance, even though the HTTP spec allows string dates, if a service has decided it will **ever** only send integer seconds in the header, would it be appropriate for the service and spec to use `integer`?

@mikekistler: Do you know the answer?

Comment (mikekistler) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: We always want the value of `retry-after` to be "delay-seconds" and not a date, and RFC 7231 says:
> A delay-seconds value is a non-negative decimal integer, representing time in seconds.

So `integer` is the appropriate type for this header value.

Furthermore, it is common that string and integer values are serialized in headers in the same way, so it might be possible to change the API description from `type: string` to `type: integer` in a non-breaking way (i.e. without changing the behavior of the service). As long as the service does not wrap the value in quotes, e.g.

```
Retry-after: "180"
```

but most services will not do this so it might be fine.

Comment (mikeharder) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: >We always want the value of retry-after to be "delay-seconds" and not a date

@mikekistler: Can you elaborate on this?  Is this an Azure guideline above the HTTP spec (which allows either)?  Why wouldn't we want to give services the option to send either seconds or a date?

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Review by marclerwick — APPROVED** [review]

File: specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json
Comment (mikekistler) [reply specification/communication/data-plane/Email/preview/2024-07-01-preview/CommunicationServicesEmail.json:49]: `Retry-after` should always specific `delay-seconds` in any response.

**Review by kagbakpem — CHANGES_REQUESTED** [review]
Let's hold off on merging this PR until the changes are reflected on the service side. 2024-07-01-preview.

**Review by kagbakpem — APPROVED** [review]

**Pull request merged**
